lib/conjur/base.rb

```diff
@@ -253,52 +271,158 @@ def with_audit_resources resource_ids
       end
     end
 
+    module MonotonicTime
+      def monotonic_time
+        Process.clock_gettime Process::CLOCK_MONOTONIC
+      rescue
+        # fall back to normal clock if there's no CLOCK_MONOTONIC
+        Time.now.to_f
+      end
+    end
+
+    module TokenExpiration
+      include MonotonicTime
+
+      # The four minutes is to work around a bug in Conjur < 4.7 causing a 404 on 
+      # long-running operations (when the token is used right around the 5 minute mark).
+      TOKEN_STALE = 4.minutes
+
+      # Don't check more often than this whether the token needs refreshing.
+      # It's just a churn to do so.
```

Comment:
What does it hurt to check 'too often'? It's not like it's expected to be in a tight loop.